File: clients/python/examples/tables-create-text-index-string-analyzer.py
from astrapy import DataAPIClient
from astrapy.info import TableTextIndexOptions

# Get an existing table
client = DataAPIClient()
database = client.get_database(
    "**API_ENDPOINT**", token="**APPLICATION_TOKEN**"
)
table = database.get_table("**TABLE_NAME**")

# Index a column
table.create_text_index(
    "**INDEX_NAME**",
    column="**TEXT_COLUMN_NAME**",
    options=TableTextIndexOptions(
        analyzer="english",
    ),
)

File: clients/java/examples/tables-create-text-index-string-analyzer.java
import com.datastax.astra.client.DataAPIClient;
import com.datastax.astra.client.core.lexical.AnalyzerTypes;
import com.datastax.astra.client.tables.Table;
import com.datastax.astra.client.tables.definition.indexes.TableTextIndexDefinition;
import com.datastax.astra.client.tables.definition.rows.Row;

public class Example {

  public static void main(String[] args) {
    // Get an existing table
    Table<Row> table =
        new DataAPIClient("**APPLICATION_TOKEN**")
            .getDatabase("**API_ENDPOINT**")
            .getTable("**TABLE_NAME**");

    // Index a column
    TableTextIndexDefinition definition =
        new TableTextIndexDefinition()
            .column("**TEXT_COLUMN_NAME**")
            .analyzer(AnalyzerTypes.ENGLISH);
    table.createTextIndex("**INDEX_NAME**", definition);
  }
}
